common/src/stack/pylib/stack/gen.py

```diff
@@ -872,3 +872,129 @@ def generate_bash(self):
+		playbook.extend(self.shellSection.generate())
+		return playbook
+
+
+class AnsibleProfileTraversor(MainTraversor):
+	def pre(self):
+		self.tasks = {
+			'install-post': [],
+			'boot-pre': [],
+			'boot-post': []
+		}
+
+	def traverse_stack_script(self, node):
+		stage  = self.getAttr(node, 'stack:stage',  default='install-post')
+		chroot = self.getAttr(node, 'stack:chroot', default='true')
+
+		# QUESTION: Do we want to ignore these stages in Ansible?
+		if stage in [ 'install-pre', 'install-pre-package' ] or not chroot == 'true':
+			return False
+
+		def tabs_to_spaces(match):
+			"""Replace each match char (a tab) with 4 spaces"""
+			return match.group(1) + ' ' * 4 * len(match.group(2))
+
+		node_id = self.getAttr(node, 'stack:id')
+		shell  = self.getAttr(node, 'stack:shell', default='/bin/bash')
+		node_file = os.path.basename(self.getAttr(node, 'stack:file'))
+		script = self.collect(node).strip()
+
+		# Make a nice name for our Ansible task
+		name = f"{stage}: Create script {node_id}"
+		if node_file:
+			name += f" from {node_file}"
+
+		# Format the script to be valid yaml by replacing tabs with spaces
+		formatted_script = []
+		last_line_blank = True
+		for line in script.splitlines():
+			# Collapse down multiple blank lines
+			if not line and last_line_blank:
+				continue
+
+			# Replace beginning tabs with spaces
+			line = re.sub(r'^( *)(\t+)', tabs_to_spaces, line)
+
+			# Strip any trailing whitespace
+			line = line.rstrip()
+
+			# Indent it for the Ansible output
+			formatted_script.append(' '*8 + line)
+
+			# Record if we were blank, so we can collapse
+			last_line_blank = not bool(line)
+
+		# Construct our Ansible task to copy over the script
+		task = [f'  - name: "{name}"']
+		task.append("    copy:")
+		task.append("      content: |")
+		task.extend(formatted_script)
+		task.append(f"      dest: /tmp/stacki_{stage}-{node_id}")
+		task.append(f"      mode: 0600")
+		self.tasks[stage].append('\n'.join(task))
+
+		# Now add a task to run it
+		name = f"{stage}: Run script {node_id}"
+		if node_file:
+			name += f" from {node_file}"
+
+		self.tasks[stage].append('\n'.join([
+			f'  - name: "{name}"',
+			f"    shell: {shell} /tmp/stacki_{stage}-{node_id}",
+			"    ignore_errors: True"
+		]))
+
+		return True
+
+	def post(self):
+		# Ansible playbook header stuff
+		self.gen.shellSection.append('\n'.join([
+			'---',
+			f'- hosts: {self.gen.attrs["hostname"]}',
+			'  tasks:'
+		]))
+
+		# Install package task
+		packages = self.gen.packageSet.getPackages()
+
+		enabled = []
+		for _, pkgs in packages['enabled'].items():
+			enabled.extend(pkgs)
+
+		if enabled:
+			task = [
+				'  - name: "Packages: Install"',
+				'    package:',
+				'      name:'
+			]
+			for package in enabled:
+				task.append(f'        - {package}')
+			task.append('      state: present')
+
+			self.gen.shellSection.append('\n'.join(task)+'\n')
+
+		# Remove package task
+		disabled = []
+		for _, pkgs in packages['disabled'].items():
```

Comment:
If you don't need the `key` this could instead iterate over `packages['disabled'].values()`